src/components/AddItem.js

```diff
@@ -19,61 +23,76 @@ const AddItemForm = props => {
 
   const addItem = e => {
     e.preventDefault();
-    console.log(emptyShoppingItem.lastPurchasedDate)
+    console.log(emptyShoppingItem.lastPurchasedDate);
+
+    const items = getShoppingList();
 
     if (enteredValue.name === '') {
       alert('Please enter an item name');
     } else {
-      firestore.collection('shoppingList').add({
-        name: enteredValue.name,
-        nextPurchase: parseInt(enteredValue.nextPurchase, 10),
-        token: localStorage.getItem('userToken'),
+      const removePunctuation = enteredValue.name.replace(
+        /(~|`|!|@|#|$|%|^|&|\*|\(|\)|{|}|\[|\]|;|:|"|'|<|,|\.|>|\?|\/|\\|\||-|_|\+|=)/g,
+        '',
+      ); // removes punctuation
+      const finalEnteredVal = removePunctuation.replace(/\s{2,}/g, ' '); // removes extra spacing
```

Comment:
Talk about regex -- wowza! Nice job here, y'all! 👏

Reply:
I'm not sure this is working as expected anymore. Would y'all mind popping back in, running some more tests and double-checking the logic here to make sure it's capturing those duplicates and popping up the modal? Here is a video of how it was failing for me: https://www.loom.com/share/49d9ef97e1f0411197567bff3630e40d